src/db/models/json/indicatorsAndCaseloads.ts

```diff
@@ -30,6 +30,28 @@ export type CaseloadOrIndicatorMetricWithValue = t.TypeOf<
   typeof METRIC_WITH_VALUE
 >;
 
+const DISAGGREGATED_LOCATIONS = t.array(
+  t.intersection([
+    t.type({
+      name: t.string,
+    }),
+    t.partial({
+      id: t.union([t.number, t.string]),
+      parent: t.type({
+        id: t.union([t.number, t.string]),
+        name: t.string,
+      }),
+    }),
+  ])
+);
+
+export const DISAGGREGATED_LOCATIONS_WITH_PCODE = t.array(
+  t.intersection([
+    DISAGGREGATED_LOCATIONS.type,
+    t.partial({ pcode: t.union([t.string, t.number, t.undefined, t.null]) }),
+  ])
+);
```

Comment:
Don't define `DISAGGREGATED_LOCATIONS_WITH_PCODE` since you need this codec as a specific case, describing data returned from your endpoint.

Instead, extract `DISAGGREGATED_LOCATIONS` to a separate variable, `export` it and define `DISAGGREGATED_LOCATIONS_WITH_PCODE` in the file you use it.